pkg/controller/controller.go

```diff
@@ -170,20 +175,19 @@ func (h *Handler) syncLocalVolumeProvider(instance *localv1.LocalVolume) error {
 		})
 	}
 	o.Status.Generations = children
-	o.Status.ObservedGeneration = &o.Generation
 	o.Status.State = operatorv1.Managed
-	condition := operatorv1.OperatorCondition{
-		Type:               operatorv1.OperatorStatusTypeAvailable,
-		Status:             operatorv1.ConditionTrue,
-		Message:            "Ready",
-		LastTransitionTime: metav1.Now(),
-	}
-	newConditions := []operatorv1.OperatorCondition{condition}
-	o.Status.Conditions = newConditions
-	err = h.apiClient.syncStatus(instance, o)
-	if err != nil {
-		klog.Errorf("error syncing status : %v", err)
-		return fmt.Errorf("error syncing status %v", err)
+	o = h.addSuccessCondition(o)
+
+	if !equality.Semantic.DeepEqual(instance.Status, o.Status) {
+		// A generation update can cause frivolous updates to LocalVolume object
+		// we are only going to update generation when something in LocalVolume object
+		// changes.
+		o.Status.ObservedGeneration = &o.Generation
+		err = h.apiClient.syncStatus(instance, o)
```

Comment:
In addition, I would put the equality check into `syncStatus`, it should be done every time something wants to update the status.